# Data Scientist Work: Demand Estimation

In [41]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score

In [42]:
df=pd.read_csv('/content/analytics_final_project/demand_est_project/air_fryers_clean_brand_year.csv')
df.head()

,category,year,brand,purchase_count,product_count,avg_price,avg_rating,compact_share,dual_basket_share,oven_style_share,rotisserie_share,window_share,market_purchases,brand_share,log_brand_share
0,air_fryers,2019,chefman,1146,10,72.963695,4.434119,1.000000,0.0,0.780977,0.243455,0.184119,15076,0.076015,-2.576826
1,air_fryers,2019,cosori,11,2,159.990000,4.581818,1.000000,0.0,0.090909,0.090909,0.000000,15076,0.000730,-7.222964
2,air_fryers,2019,cuisinart,1616,22,229.465274,4.481312,0.993812,0.0,0.889851,0.000000,0.000000,15076,0.107190,-2.233150
3,air_fryers,2019,dash,3011,19,55.176333,4.390767,1.000000,0.0,0.973431,0.000000,0.000000,15076,0.199721,-1.610832
4,air_fryers,2019,gowise usa,4405,45,83.575551,4.552259,0.999773,0.0,0.129398,0.128490,0.000000,15076,0.292186,-1.230364


In [43]:
y=df['log_brand_share']

brand_dummies=pd.get_dummies(df['brand'],drop_first=True, prefix="brand", dtype=int)
year_dummies=pd.get_dummies(df['year'],drop_first=True, prefix="year", dtype=int)
feature_cols = [
    'compact_share',
    'dual_basket_share',
    'oven_style_share',
    'rotisserie_share',
    'window_share']

X=pd.concat([df[['avg_price','avg_rating']+ feature_cols],brand_dummies,year_dummies],axis=1)

In [44]:
model=LinearRegression()
model.fit(X,y)

predicted_log_share=model.predict(X)
r2=r2_score(y,predicted_log_share)

coef_table = pd.DataFrame({
    "feature": X.columns,
    "coefficient": model.coef_})

print("R-squared:", r2)
coef_table

R-squared: 0.763453950091436


,feature,coefficient
0,avg_price,-0.037668
1,avg_rating,0.287517
2,compact_share,9.815304
3,dual_basket_share,-9.509686
4,oven_style_share,1.941774
5,rotisserie_share,-5.674054
6,window_share,12.880298
7,brand_cosori,2.551946
8,brand_cuisinart,6.422436
9,brand_dash,0.176655


### Questions

1. What is the estimated price coefficient, $\hat{\beta}_{price}$?


In [45]:
coef_table.loc[coef_table['feature']=='avg_price']

,feature,coefficient
0,avg_price,-0.037668


The estimated price coefficient is -0.037668.

2. Is it negative? Why is that important?


The price coefficient is negative, which means that as the price increases, the demand decreases. This suggests that consumers are less likely to buy more expensive air fryers. This is important for companies selecting the appropriate price for their product, since higher prices can lead to lower demand.

3. Which product features are associated with higher demand?


In [46]:
coef_table.sort_values(by='coefficient', ascending=False)

,feature,coefficient
6,window_share,12.880298
2,compact_share,9.815304
8,brand_cuisinart,6.422436
12,brand_ninja,5.838705
11,brand_instant_pot,4.626260
10,brand_gowise usa,3.938996
14,brand_oster,3.928074
13,brand_nuwave,3.544883
7,brand_cosori,2.551946
4,oven_style_share,1.941774


The two strongest positive predictors of demand are window_share (12.88) and compact_share (9.82). This suggests that air fryers that contain windows or those that are compact are the most valued.

4. Which brand dummy coefficients are largest? Remember that these are interpreted relative to the dropped brand.





Cuisinart (6.42) has the largest dummy coefficient relative to the omitted brand. Ninja (5.84) and Instant Pot (4.63) also have a strong positive effect. All listed brands have positive coefficients, meaning that they outperform the reference brand in terms of demand, holding price, rating, and product features constant.

5. Which year dummy coefficients are largest? Remember that these are interpreted relative to the dropped year.


The year 2020 has the highest dummy coefficient (0.12), meaning that demand for air fryers was highest in 2020 relative to the dropped year. The second highest year coefficient is 2021, with a value of 0.042.

6. What is the model's $R^2$?

The model's R^2 is 0.763, meaning 76% of the variation in log brand share is explained by the independent variables. This indicates an overall strong explanatory power and suggests that price, product features, brand effects, and time effects are key drivers of demand in this market.